sbt-cross/src/main/scala/sbtcross/CrossProject.scala

```diff
@@ -0,0 +1,383 @@
+          }
+
+        project.dependsOn(dependencies: _*)
+      })(p => p)
+
+    new CrossProject(crossType, platforms, projectsDepending)
+  }
+
+  def disablePlugins(ps: AutoPlugin*): CrossProject =
+    transform(_.disablePlugins(ps: _*))
+
+  def enablePlugins(ns: Plugins*): CrossProject =
+    transform(_.enablePlugins(ns: _*))
+
+  def in(dir: File): CrossProject =
+    new CrossProject(crossType, platforms, projects.map {
+      case (platform, project) =>
+        platform -> project.in(crossType.platformDir(dir, platform))
+    })
+
+  def overrideConfigs(cs: Configuration*): CrossProject =
+    transform(_.overrideConfigs(cs: _*))
+
+  def settingSets(select: AddSettings*): CrossProject =
+    transform(_.settingSets(select: _*))
+
+  def settings(ss: Def.SettingsDefinition*): CrossProject =
+    transform(_.settings(ss: _*))
+
+  override def toString(): String =
+    projects.map {
+      case (platform, project) =>
+        s"${platform.identifier} = $project"
+    }.mkString("CrossProject(", ",", ")")
+
+  private def transform(f: Project => Project): CrossProject =
+    new CrossProject(crossType, platforms, projects.mapValues(f))
+
+  private def join[K, A, B, Z](m1: Map[K, A], m2: Map[K, B])(f: (A, B) => Z)(
+      da: A => Z): Map[K, Z] = {
+    (m1.keySet -- m2.keySet).map(k => k -> da(m1(k))).toMap ++ // missing in m2
+      m1.flatMap {
+        case (k, a) =>
+          m2.get(k).map(b => Map(k -> f(a, b))).getOrElse(Map.empty[K, Z])
+      }
+  }
+}
+
+object CrossProject {
+  final case class Builder(id: String, base: File, platforms: Platform*) {
+    def crossType(crossType: CrossType): CrossProject =
+      CrossProject(id, base, crossType, platforms: _*)
+  }
+
+  def apply(id: String,
+            base: File,
+            crossType: CrossType,
+            platforms: Platform*): CrossProject = {
+    def sharedSrcSettings(crossType: CrossType) = {
+      def makeCrossSources(sharedSrcDir: Option[File],
+                           scalaBinaryVersion: String,
+                           cross: Boolean): Seq[File] = {
+        sharedSrcDir match {
+          case Some(dir) =>
+            if (cross)
+              Seq(dir.getParentFile / s"${dir.name}-$scalaBinaryVersion", dir)
+            else
+              Seq(dir)
+          case None => Seq()
+        }
+      }
+
+      Seq(
+        unmanagedSourceDirectories in Compile ++= {
+          makeCrossSources(crossType.sharedSrcDir(baseDirectory.value, "main"),
+                           scalaBinaryVersion.value,
+                           crossPaths.value)
+        },
+        unmanagedSourceDirectories in Test ++= {
+          makeCrossSources(crossType.sharedSrcDir(baseDirectory.value, "test"),
+                           scalaBinaryVersion.value,
+                           crossPaths.value)
+        }
+      )
+    }
+
+    val shared = sharedSrcSettings(crossType)
+
+    val projects =
+      platforms
+        .map(
```

Comment:
In general, higher-order-functions should be called with `{}` unless the inner lambda fits on one line, on the same line as the call itself.